### PR TITLE
Separate descriptor set for textures in vulkan

### PR DESCRIPTION
**Pull request by apjagdale**
Added set 1 of descriptor for textures so that now they do not conflict with other uniforms in the shader.
The start of textures is kept from binding 10 in order to match with its texcoord location.

GearVRf DCO signed off by: Abhijit Jagdale a.jagdale@samsung.com

### CODE REVIEW

File: GVRf/Framework/framework/src/main/jni/vulkan/vulkan_render_data.cpp [hunk 1]
```diff
@@ -35,17 +35,23 @@ void VulkanRenderData::render(Shader* shader, VkCommandBuffer cmdBuffer, int cur
...
-    VkDescriptorSet descriptorSet = getDescriptorSet(curr_pass);
     //bind out descriptor set, which handles our uniforms and samplers
     if(static_cast<VulkanShader*>(shader)->isDepthShader()){
         vkCmdBindDescriptorSets(cmdBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS,
                                 Vkshader->getPipelineLayout(), 0, 1,
-                                &rp->m_descriptorSet, 0, NULL);
+                                &rp->m_descriptorSet[0], 0, NULL);
+
     }
     else if (!isDescriptorSetNull(curr_pass)) {
```
Comment (roshanch) [inline GVRf/Framework/framework/src/main/jni/vulkan/vulkan_render_data.cpp]: please change this for multiple descriptor sets.

**Comment by NolaDonato** [comment]
Do we just have two descriptor sets or can we have more?

**Comment by apjagdale** [comment]
@NolaDonato Currently we have 2. We can have more.

**Comment by liaxim** [comment]
gvr-accessibility on Note 8 with Vulkan enabled crashes for me on entering the options scene:
```
08-02 08:26:00.558 22710 22804 I Adreno  : Program compilation failed
08-02 08:26:00.558 22710 22804 I Adreno  : Info log: Error: Bind group info mismatches the shader source for symbol u_texture
08-02 08:26:00.558 22710 22804 I Adreno  : 
08-02 08:26:00.558 22710 22804 E gvrf    : VK_CHECK Failure
```

**Comment by liaxim** [comment]
Ok then. Planning to merge this on Friday.

**Pull request merged**